### DISCUSSION TITLE
Firebase storage  : Error: [storage/unknown] An unknown error has occurred. (IOS)

### DISCUSSION THREAD

**Discussion by AchrafBn** [invertase/react-native-firebase#6154] [Q&A]
After uploaded image to cloud storage  I'm getting this error , Any clue please  ?

Sample code 
```

 const onUpload = async (
    uri
  ) => {

    try {
      const file = await FileSystem.readAsStringAsync(uri, {
        encoding: FileSystem.EncodingType.Base64,
      });
      const ref = await storage().ref().child(currentUser.uid);
      let snap = await ref.putString(file, "base64");
      const url = await snap.ref.getDownloadURL();
    } catch (e) {
      console.log("Error", e);
    }
  };
```
**Dependencies**

    "@react-native-firebase/app": "14.7.0",
    "@react-native-firebase/auth": "14.7.0",
    "@react-native-firebase/firestore": "14.7.0",
    "@react-native-firebase/storage": "^14.7.0",
    "react-native": "0.64.3",

**Console** 
```
[Unhandled promise rejection: Error: [storage/unknown] An unknown error has occurred.]
at node_modules/@react-native-firebase/storage/lib/StorageTask.js:152:22 in get__then
- ... 10 more stack frames from framework internals
```
**Screenshot**

<img width="333" alt="Screenshot 2022-03-24 at 14 22 50" src="https://user-images.githubusercontent.com/11798130/159925514-4321e9e3-ee83-49fa-9c1d-d98b16f38c15.png">

**Comment by mikehardy** [comment]
Hi there!
No idea, sorry. You should run the project from Xcode and watch the console. An internal error is typically accompanied by some messaging from firebase-ios-sdk in the lower-level logs

**Comment by marcfrankel** [comment]
I am having this exact same issue. No clue as to what is causing it yet, but please keep me updated!

**Comment by itodevio** [comment]
Any updates on this? I'm getting this error... the first time I install the app I can upload any image once... then it keeps throwing this error

**Reply by death14stroke** [reply]
Hey anyone found any solution yet? I am facing the same issue on ios simulator - 

`Error: [storage/unknown] An unknown error has occurred.`

**Comment by AchrafBn** [comment]
I didn't get this error when testing on IOS device, but it shows only on ios simulator

**Reply by Gautham495** [reply]
> I didn't get this error when testing on IOS device, but it shows only on ios simulator

This the true answer unfortunately

**Comment by sahuyash1103** [comment]
I also have the same issue but in android?

**Comment by Pramod058** [comment]
This is because of the TEST MODE while initializing firebase database. It expires in 1 month and throw this error.

**Reply by lsps9150414** [reply]
@Pramod058 could you elaborate what `TEST MODE` is and how can we config it to avoid the issue?

**Reply by lsps9150414** [reply]
Hey @Pramod058 it'll be really helpful if you could share more details on this 🙏🏻

**Comment by bibhuti9** [comment]
I re-build my project in x-code.

it worked for me

**Comment by zahid502** [comment]
Erase the device content and settings and then build, it worked for me.
It's due to simulator

<img width="347" alt="Screenshot 2023-09-27 at 6 33 21 PM" src="https://github.com/invertase/react-native-firebase/assets/115239786/94698d93-0532-4804-826d-ebe9d11b63be">

**Comment by younisrahman** [comment]
Try with a small-sized image.

**Comment by markwitt1** [comment]
Any updates on this? It happens to me quite randomly. What would be possible reasons?

**Reply by mikehardy** [reply]
I've never had good luck guessing at reasons people state happen to them randomly, but without specifying any details on reproduction. Have you?

https://stackoverflow.com/help/how-to-ask
https://stackoverflow.com/help/minimal-reproducible-example

**Comment by Gone-M** [comment]
If you are trying with web module yes you may get this error but if you try to install with firebase sdk the problem will be solved! I had the same problem and this problem occurs a lot with web module.

**Comment by devlprkhan** [comment]
+1

**Comment by ics-ankit** [comment]
Same issue faced when uploading the image on ios simulator  (storage firebase) -> used image picker for expo-image-picker and camera same library
-  [storage/unknown] An unknown error has occurred.
on Android working fine

**Comment by ics-ankit** [comment]
@AchrafBn @mikehardy @Pramod058 @death14stroke @devlprkhan @Gone-M 
any solution ?

**Comment by lucasmendesdeliver** [comment]
Same issue here
